zetaclient/chains/evm/observer/observer.go

```diff
@@ -508,6 +558,7 @@ func (ob *Observer) PostGasPrice() error {
 }
 
 // TransactionByHash query transaction by hash via JSON-RPC
+// TODO(revamp): move to a tx file
```

Comment:
Maybe `update this method as a pure RPC method that takes two parameters and move to upper package to file rpc.go`